source/extensions/filters/listener/http_inspector/http_inspector.cc

```diff
@@ -132,31 +153,26 @@ ParseState Filter::parseHttpHeader(absl::string_view data) {
     protocol_ = "HTTP/2";
     return ParseState::Done;
   } else {
-    const size_t pos = data.find_first_of("\r\n");
-    if (pos != absl::string_view::npos) {
-      const absl::string_view request_line = data.substr(0, pos);
-      const std::vector<absl::string_view> fields =
-          absl::StrSplit(request_line, absl::MaxSplits(' ', 4));
-
-      // Method SP Request-URI SP HTTP-Version
-      if (fields.size() != 3) {
-        ENVOY_LOG(trace, "http inspector: invalid http1x request line");
-        // done(false);
-        return ParseState::Error;
-      }
+    ssize_t rc =
+        http_parser_execute(&parser_, &settings_, data.data() + read_, data.length() - read_);
+    ENVOY_LOG(trace,
+              "http inspector: http_parser parsed {} chars, error code: {}, parser state: {}", rc,
+              HTTP_PARSER_ERRNO(&parser_), parser_.state);
+
+    // Errors in parsing HTTP.
+    if (HTTP_PARSER_ERRNO(&parser_) != HPE_OK && HTTP_PARSER_ERRNO(&parser_) != HPE_PAUSED) {
+      return ParseState::Error;
+    }
 
-      if (http1xMethods().count(fields[0]) == 0 || httpProtocols().count(fields[2]) == 0) {
-        ENVOY_LOG(trace, "http inspector: method: {} or protocol: {} not valid", fields[0],
-                  fields[2]);
-        // done(false);
-        return ParseState::Error;
+    // state 18 is the s_start_req. When the header is parsed, the http_parser will reset the state
+    // to s_start_req. state 42 is s_req_http_end
+    if ((parser_.state == 18 && message_complete_) || parser_.state >= 42) {
```

Comment:
Why is not message_complete_ sufficient here in addition to state >= 42.  What other state might it be in with message_complete_ == true?  Comment.

Reply:
Yes, I think `message_complete_` is sufficient. I updated PR to check `message_complete_` only.